plugins/pipelines/test/models/stage_test.rb

```diff
@@ -0,0 +1,86 @@
+require_relative '../test_helper'
+
+describe Stage do
+  let(:project) { Project.create!(name: 'foo', repository_url: 'random') }
+  let(:stage1) { Stage.create!(project: project, name: 'stage1') }
+  let(:stage2) { Stage.create!(project: project, name: 'stage2') }
+  let(:stage3) { Stage.create!(project: project, name: 'stage3') }
+
+  before do
+    Project.any_instance.stubs(:valid_repository_url).returns(true)
+    project.stages = [stage1, stage2, stage3]
+  end
+
+  describe '#next_stage' do
+    it 'returns next created stage if no pipeline set' do
+      stage2.next_stage.id.must_equal stage3.id
+    end
+
+    it 'returns next stage in pipeline if set' do
+      stage2.update!(next_stage_ids: [ stage1.id, stage3.id ])
+      stage2.next_stage.id.must_equal stage1.id
+    end
+  end
+
+  describe '#production?' do
+    it 'returns false if no pipeline set and not marked production' do
+      stage1.deploy_groups = [ deploy_groups(:pod100) ]
+      stage1.production?.must_equal false
+    end
+
+    it 'returns true if no pipeline set and marked production' do
+      stage1.deploy_groups = [ deploy_groups(:pod1) ]
```

Comment:
let(:production) + let(:staging) might make this a bit more readable, not everyone might know what pod1 / pod100 means :)